### DISCUSSION TITLE
HasManyThrough Syntax

### DISCUSSION THREAD

**Discussion by popovicieduard** [adonisjs/core#1295] [Help] [answered]
Hello, I was wondering what is the syntax for HasManyThrough ?

if the syntax for HasMany is:

```
  @hasMany(() => Token)
  public tokens: HasMany<typeof Token>
```

syntax for HasManyThrough is ? Couldn't find anything in the docs.

**Comment by iammrsea** [comment, marked as answer]
For instance, you have an app that has the following models: User, Restaurant and Dish. Here, a user hasOne restaurant and that restaurant has many dishes associated with it. Invariably, same user has many dishes through restaurant. This is how you could establish the relationships:

```javascript
//Dish.ts
@belongsTo(()=>Restaurant)
public restaurant: BelongsTo<typeof Restaurant>;

//Restaurant.ts
@belongsTo(()=>User)
public user: BelongsTo<typeof User>;

@hasMany(()=>Dish)
public dishes: HasMany<typeof Dish>;

//User.ts
@hasOne(()=>Restaurant)
public restaurant: HasOne<typeof Restaurant>;

@hasManyThrough([()=>Dish, ()=>Restaurant])
public dishes: HasManyThrough<typeof Dish>;
```
